controller/hybridgateway/controller.go

```diff
@@ -129,6 +131,22 @@ func (r *HybridGatewayReconciler[t, tPtr]) Reconcile(ctx context.Context, req ct
 	gvk := obj.GetObjectKind().GroupVersionKind()
 	log.Debug(logger, "Reconciling object", "Group", gvk.Group, "Kind", gvk.Kind)
 
+	// Handle deletion and finalizer cleanup
+	if obj.GetDeletionTimestamp() != nil {
+		return r.handleDeletion(ctx, logger, obj, rootObj)
```

Comment:
Can we handle deletion gracefully? Something similar to https://github.com/Kong/kong-operator/blob/aa12e0421b12a21a4436b2c163fcf37ddf29bd7e/controller/gateway/controller_cleanup.go#L50-L57

Reply:
After discussion we decided to not deal with the grace period for HTTPRoute deletion.